### PR TITLE
Initial file structure for CLI.

### PR DESCRIPTION
**Pull request by albertpurnama**
> I'm intentionally not linking the GH issue because it doesn't solve the issue yet. This is merely a quick vibe check on the folder/cmd structure.

Initial implementation of llmgo's CLI.

Here's a demo video

https://github.com/joshcarp/llm.go/assets/14824254/6e8be9c7-8b83-4f36-bbe6-799eeeab273d

I'm opening this PR early to get feedback on the structure of the command itself.

### CODE REVIEW

**Comment by joshcarp** [comment]
Works better than I expected, I haven't looked at the parameter/tokenizer binary files but what do you think about having them load directly from huggingface?

We'll put it in another issue, but having it load from here: https://huggingface.co/joshcarp/llm.go/tree/main for example would mean that we could unlock the potential to `go install` it as a CLI.

File: command.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/*
+Copyright © 2024 NAME HERE <EMAIL ADDRESS>
+*/
```
Comment (joshcarp) [inline command.go:3]: Probably don't need this

Comment (joshcarp) [inline command.go]: Remove this stuff

File: command.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+// gpt2Cmd represents the gpt2 command
+var gpt2Cmd = &cobra.Command{
+	Use:   "gpt2",
+	Short: "A brief description of your command",
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
```
Comment (joshcarp) [inline command.go]: Let's get gpt to write this 😂

File: command.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	Run: func(cmd *cobra.Command, args []string) {
+		for {
+			fmt.Printf(">>> ")
+			inputReader := bufio.NewReader(os.Stdin)
+
+			input, err := inputReader.ReadString('\n')
+			if err != nil {
+				if err == io.EOF {
+					fmt.Println("\nExiting command. Thanks for using llmgo!")
+					os.Exit(0)
```
Comment (joshcarp) [inline command.go]: I think you can just `break` or `return` here; `os.Exit` shouldn't really be called in anything that isn't the main function

File: command.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	Short: "A brief description of your command",
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	// Run: func(cmd *cobra.Command, args []string) {
+	// 	fmt.Println("run called")
+	// },
```
Comment (joshcarp) [inline command.go]: This text can be replaced/deleted

File: command.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	Long: `A longer description that spans multiple lines and likely contains
+examples and usage of using your application. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	// Uncomment the following line if your bare application
+	// has an action associated with it:
+	// Run: func(cmd *cobra.Command, args []string) { },
+}
```
Comment (joshcarp) [inline command.go]: Same here

**Review by joshcarp — COMMENTED** [review]
Nice work, it works a lot better than I was expecting it to, which is interesting.

I haven't looked into the chat templates that the GPT-2 uses, and we might want to think about how it handles that in the future incase other tokenizers/weights are used

Just some cleanup to do and this will be good to merge

**Comment by albertpurnama** [comment]
Resolved @joshcarp !

**Comment by joshcarp** [comment]
Thanks @albertpurnama!

**Pull request merged**